src/test/kotlin/com/ampnet/identityservice/service/AutoInvestQueueServiceTest.kt

```diff
@@ -0,0 +1,610 @@
+@Import(TestSchedulerConfiguration::class)
+
+class AutoInvestQueueServiceTest : TestBase() {
+
+    private var address1 = "0xbdD53fE8b8c2359Ed321b6ef00908fb3e94D0aF7"
+    private val address2 = "0xbdD53fE8b8c2359Ed321b6ef00908fb3e94D0aF8"
+    private val address3 = "0xbdD53fE8b8c2359Ed321b6ef00908fb3e94D0aF9"
+    private val campaign1 = "0xbdD53fE8b8c2359Ed321b6ef00908fb3e94D0aFa"
+    private val campaign2 = "0xbdD53fE8b8c2359Ed321b6ef00908fb3e94D0aFb"
+    private val campaign3 = "0xbdD53fE8b8c2359Ed321b6ef00908fb3e94D0aFc"
+    private val hash = "0x6f7dea8d5d98d119de31204dfbdc69bb1944db04891ad0c45ab577da8e6de04a"
+    private val chainId = Chain.MATIC_TESTNET_MUMBAI.id
+
+    @Autowired
+    private lateinit var databaseCleanerService: DatabaseCleanerService
+
+    @Autowired
+    private lateinit var autoInvestTaskRepository: AutoInvestTaskRepository
+
+    @Autowired
+    private lateinit var autoInvestTransactionRepository: AutoInvestTransactionRepository
+
+    @Autowired
+    private lateinit var applicationProperties: ApplicationProperties
+
+    @Autowired
+    private lateinit var zonedDateTimeProvider: ZonedDateTimeProvider
+
+    @Autowired
+    private lateinit var uuidProvider: UuidProvider
+
+    @MockBean
+    private lateinit var blockchainService: BlockchainService
+
+    @Autowired
+    private lateinit var autoInvestQueueService: AutoInvestQueueService
+
+    @Autowired
+    private lateinit var autoInvestQueueScheduler: ManualFixedScheduler
+
+    @BeforeEach
+    fun beforeEach() {
+        databaseCleanerService.deleteAllAutoInvestTasks()
+        databaseCleanerService.deleteAllAutoInvestTransactions()
+    }
+
+    @AfterEach
+    fun afterEach() {
+        databaseCleanerService.deleteAllAutoInvestTasks()
+        databaseCleanerService.deleteAllAutoInvestTransactions()
+    }
+
+    @Test
+    fun mustCorrectlyCreateAutoInvestTask() {
+        var response: AutoInvestResponse? = null
+        suppose("Auto-invest task creation is requested") {
+            response = autoInvestQueueService.createOrUpdateAutoInvestTask(
+                address = address1,
+                campaign = campaign1,
+                chainId = chainId,
+                request = AutoInvestRequest(amount = BigInteger.valueOf(100L))
+            )
+        }
+
+        verify("Auto-invest task is created") {
+            assertThat(response).isNotNull()
+            assertThat(response).isEqualTo(
+                AutoInvestResponse(
+                    walletAddress = address1,
+                    campaignAddress = campaign1,
+                    amount = BigInteger.valueOf(100L)
+                )
+            )
+
+            val databaseTask = autoInvestTaskRepository.findByUserWalletAddressAndCampaignContractAddressAndChainId(
+                userWalletAddress = address1,
+                campaignContractAddress = campaign1,
+                chainId = chainId
+            )
+
+            assertThat(databaseTask).isNotNull()
```

Comment:
You could escape using nullable task below using:
```suggestion
            ) ?: fail("Missing database task);
```

Reply:
This doesn't work, because Kotlin doesn't see `fail` as an exception being thrown so it still thinks that `databaseTask` can be `null`.